### PR TITLE
Fix - Osmosis tests

### PR DESCRIPTION
**Pull request by MissingNO57**
## Proposed changes

- Moved Osmosis tests to a new group named third-party tests
- Upgraded Osmosis test-net to v5
https://docs.osmosis.zone/networks/

- Currently, there is no way to use the Osmosis faucet, so we will be using the well-funded account `osmo1ya7ymmhq3xq94parz46w5jhf2v83hqd83aac8e`

## Issues

Links to any issues resolved.

## Types of changes

What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to stop working as expected)
- [ ] Something else (e.g. tests, scripts, example, deployment, infrastructure, ...)

## Checklist

_Put an `x` in the boxes that apply._

- [x] I have read the [CONTRIBUTING](https://github.com/fetchai/cosmpy/blob/main/CONTRIBUTING.md) document.
- [x] I have based my branch, and I am making a pull request against, the `main` branch.
- [x] Checks and tests pass locally.

### If applicable

- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] I have checked that code coverage does not decrease.
- [x] I have added/updated the documentations.

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did, what alternatives you considered, etc.

### CODE REVIEW

File: .github/workflows/workflow.yml [hunk 1]
```diff
@@ -107,3 +107,25 @@ jobs:
...
+    timeout-minutes: 30
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-python@v4
+        with:
+          python-version: '${{ matrix.python-version }}'
+      - name: Install dependencies
+        run: pip install tox==3.25.1
+      - name: Third Party Tests
+        run: |
```
Comment (pbukva) [inline .github/workflows/workflow.yml:130]: Try this (based on [this suggestion](https://github.com/actions/runner/issues/2347#issuecomment-1367587319)):

```suggestion
        continue-on-error: true
        run: |
```

File: tests/conftest.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    """
+    Modify items collection for pytest.
+
+    :param items: list of items
+    """
+    # Remove third party tests from integration tests
+    for item in items:
+        if "third_party" in item.nodeid:
+            if "integration" in item.keywords:
+                item.add_marker(pytest.mark.skip(reason="Skipped in integration tests"))
```
Comment (pbukva) [inline tests/conftest.py:37]: Is this newly added function `pytest_collection_modifyitems(...)` called/used anywhere?

**Review by pbukva — APPROVED** [review]
LGTM (have a look at my comment - I will leave it upon you)

File: tests/third_party/osmosis_testnet/net_config.py [hunk 3]
```diff
@@ -18,42 +18,40 @@
...
-        for i in range(10):
-            if ledger.query_bank_balance(wallet.address()) > 0:
-                break
-            sleep(i * 2)
-        else:
-            raise Exception("fail to topup")
+
+        funded_wallet = LocalWallet.from_unsafe_seed(FUNDED_PK, prefix=PREFIX)
+        ledger.send_tokens(
+            wallet.address(), 100, DENOM, funded_wallet
```
Comment (jrriehl) [inline tests/third_party/osmosis_testnet/net_config.py:56]: Just wondering how much funds does the `FUNDED_PK` have? In any case, we could probably send the minimum amount here, to maximize the amount of tests we can do from the funds available. Or another possibility would be to send back and forth between two funded wallets.

Comment (pbukva) [reply tests/third_party/osmosis_testnet/net_config.py:56]: is there any possibility to make these osmoses tests independent from external testnet?
For example by creating a docker setup (e.g. via docker compose thing, or similar approach) which can be started as local node and would substitute external testnet. That way, these tests will be self-sufficient = can be run independently from external testnet.

File: tests/conftest.py (hunk 2 above)
Comment (pbukva) [reply tests/conftest.py:37]: Never mind, the `pytest_collection_modifyitems(...)` is predefined [pytest hook](https://docs.pytest.org/en/7.1.x/how-to/writing_hook_functions.html), so it should be picked up automatically.

File: Makefile [hunk 4]
```diff
@@ -56,6 +56,11 @@ unit-test:
 integration-test:
 	coverage run -m pytest $(COSMPY_TESTS_DIR) --doctest-modules -m "integration"
 
+# Run all third party tests
+.PHONY: third-party-test
+third-party-test:
+	coverage run -m pytest $(COSMPY_TESTS_DIR) --doctest-modules -m "third party"
```
Comment (pbukva) [inline Makefile:62]: This comment seems to failing during test run - the third [see this line](https://github.com/fetchai/cosmpy/actions/runs/5069951654/jobs/9104289269?pr=357#step:5:62) in test run.

Comment (pbukva) [reply Makefile:62]: it kind of feels like that `"third_party"` should be there instead of `"third party"`.

Comment (MissingNO57) [reply Makefile:62]: It is the same as "not integration" few lines above
```
markers = [
    "integration: marks tests as integration (deselect with '-m \"not integration\"')",
```

File: tests/third_party/osmosis_testnet/net_config.py (hunk 3 above)
Comment (MissingNO57) [reply tests/third_party/osmosis_testnet/net_config.py:56]: We can do it in next PR.

Currently tests are optional and funded account gets Coin(amount='100000000', denom='uosmo')

**Review by jrriehl — APPROVED** [review]

File: tests/third_party/osmosis_testnet/net_config.py [hunk 5]
```diff
@@ -18,42 +18,44 @@
...
-
+from cosmpy.aerial.client import LedgerClient
 from cosmpy.aerial.config import NetworkConfig
+from cosmpy.aerial.wallet import LocalWallet
+
 
+# Network config
+PREFIX = "osmo"
+COIN = "uosmo"
+GAS_LIMIT = 120000
```
Comment (MissingNO57) [inline tests/third_party/osmosis_testnet/net_config.py:29]: This is here to override inherited params - I tried to init with super.init but pytests weren't working with it. - But I agree this is not the best way of doing it.

File: tests/third_party/osmosis_testnet/net_config.py [hunk 6]
```diff
@@ -18,42 +18,44 @@
...
+    url="grpc+https://grpc.osmotest5.osmosis.zone:443",
     fee_minimum_gas_price=1,
-    fee_denomination="uosmo",
-    staking_denomination="uosmo",
+    fee_denomination=COIN,
+    staking_denomination=COIN,
 )
+
 DEFAULT_TIMEOUT = 60.0
+# PK for funded wallet osmo1ya7ymmhq3xq94parz46w5jhf2v83hqd83aac8e
```
Comment (MissingNO57) [inline tests/third_party/osmosis_testnet/net_config.py:40]: If the wallet will be funded enough we can actually finish these tests....

File: tests/third_party/osmosis_testnet/net_config.py [hunk 7]
```diff
@@ -18,42 +18,44 @@
...
-            if ledger.query_bank_balance(wallet.address()) > 0:
-                break
-            sleep(i * 2)
-        else:
-            raise Exception("fail to topup")
+
+        funded_wallet = LocalWallet.from_unsafe_seed(FUNDED_PK, prefix=PREFIX)
+        ledger.send_tokens(
+            wallet.address(), amount, COIN, funded_wallet
+        ).wait_to_complete()
```
Comment (MissingNO57) [inline tests/third_party/osmosis_testnet/net_config.py:61]: Replaced facuet API which doesn't exist for new testnet with requesting funds from the well-funded account.

**Review by pbukva — APPROVED** [review]

**Pull request merged**